File: LDC.Atlas.Services.Execution/Application/Commands/AllocateSectionCommand.cs
using LDC.Atlas.Services.Execution.Entities;
using MediatR;

[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("LDC.Atlas.IntegrationTest")]
namespace LDC.Atlas.Services.Execution.Application.Commands
{
    public class AllocateSectionCommand : IRequest<long>
    {
        internal string Company { get; set; } // internal to avoid the exposure in Swagger

        public long? DataVersionId { get; set; }

        public long SectionId { get; set; }

        public long AllocatedSectionId { get; set; }

        public decimal Quantity { get; set; }

        public ShippingType ShippingType { get; set; }

        public AllocationType AllocationSourceType { get; set; }

        public AllocationType AllocationTargetType { get; set; }
        public int ContractInvoiceTypeId { get; set; }
    }
}

File: LDC.Atlas.WebApp/Client/app/shared/services/execution/dtos/allocate-section-command.ts
import { AllocationType } from '../../../../shared/enums/allocation-type.enum';
import { ShippingType } from '../../../../shared/enums/shipping-type-enum';
import { SplitType } from '../../../../shared/enums/split-type.enum';
import { ContractInvoiceType } from '../../../../shared/enums/contract-invoice-type.enum';

export class AllocateSectionCommand {
    sectionId: number;
    dataVersionId?: number;
    allocatedSectionId: number;
    shippingType: ShippingType;
    quantity: number;
    sourceQuantity: number;
    targetQuantity: number;
    sectionReference: string;
    allocatedSectionReference: string;
    splitType: SplitType;
    allocationSourceType: AllocationType;
    allocationTargetType: AllocationType;
    contractInvoiceTypeId: ContractInvoiceType;
}
